src/context/EntityContext.tsx

```diff
@@ -27,4 +28,27 @@ const useEntityType = () => {
     return context;
 };
 
-export { EntityContextProvider, useEntityType };
+const useEntityTypeTranslated = () => {
+    const intl = useIntl();
+
+    const entityTypeValue = useEntityType();
+
+    return useMemo(() => {
+        switch (entityTypeValue) {
+            case 'capture':
+                return intl.formatMessage({ id: 'terms.capture' });
+                break;
+            case 'materialization':
+                return intl.formatMessage({ id: 'terms.materialization' });
+                break;
+            case 'collection':
+                return intl.formatMessage({ id: 'terms.transformation' });
+                break;
```

Comment:
Either the hook should be renamed to indicate the limited scope to which it applies or a comment should be left behind indicated its intended scope.